client/blocks/login/login-form.jsx

```diff
@@ -231,28 +256,32 @@ export class LoginForm extends Component {
 								<FormInputValidation isError text={ requestError.message } />
 							) }
 
-						<label htmlFor="password" className="login__form-userdata-username">
-							{ this.props.translate( 'Password' ) }
-						</label>
-
-						<FormPasswordInput
-							autoCapitalize="off"
-							autoComplete="off"
-							className={ classNames( 'login__form-userdata-username-password', {
-								'is-error': requestError && requestError.field === 'password',
-							} ) }
-							onChange={ this.onChangeField }
-							id="password"
-							name="password"
-							ref={ this.savePasswordRef }
-							value={ this.state.password }
-							{ ...isDisabled }
-						/>
-
-						{ requestError &&
-							requestError.field === 'password' && (
-								<FormInputValidation isError text={ requestError.message } />
-							) }
+						{ ( linkingSocialUser || isRegularAccount( accountType ) ) && (
```

Comment:
We could probably refactor this condition into a function
```js
isPasswordVisible() {
    return this.props.socialAccountIsLinking || isRegularAccount( this.props.accountType )
}
```

Reply:
Makes sense indeed, I'll do that.